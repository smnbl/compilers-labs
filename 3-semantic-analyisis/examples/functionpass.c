// RUN-WITH-ARGS: -dump-function-table
int f1() {}
float f2(int x, float y, string z) {}
