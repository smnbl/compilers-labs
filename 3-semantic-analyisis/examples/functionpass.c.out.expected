└── Program <7>
    ├── FuncDecl: returnType = 'int', name = 'f1' <1>
    │   └── CompoundStmt <0>
    └── FuncDecl: returnType = 'float', name = 'f2' <6>
        ├── VarDecl: type = 'int', name = 'x' <2>
        ├── VarDecl: type = 'float', name = 'y' <3>
        ├── VarDecl: type = 'string', name = 'z' <4>
        └── CompoundStmt <5>
Function table:
f1                  i64 ()
f2                  float (i64, float, i8*)
